modules/kafka/kafka.go

```diff
@@ -125,12 +152,78 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
 	return &KafkaContainer{Container: container, ClusterID: clusterID}, nil
 }
 
-func WithClusterID(clusterID string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) error {
-		req.Env["CLUSTER_ID"] = clusterID
+func trimListeners(listeners []KafkaListener) {
+	for i := 0; i < len(listeners); i++ {
+		listeners[i].Name = strings.ToUpper(strings.Trim(listeners[i].Name, " "))
+		listeners[i].Ip = strings.Trim(listeners[i].Ip, " ")
+		listeners[i].Port = strings.Trim(listeners[i].Port, " ")
+	}
+}
 
-		return nil
+func validateListeners(listeners []KafkaListener) error {
+	var ports map[string]bool = make(map[string]bool, len(listeners)+2)
+	var names map[string]bool = make(map[string]bool, len(listeners)+2)
+
+	// check for default listeners
+	ports["9094"] = true
+	ports["9093"] = true
+
+	// check for default listeners
+	names["CONTROLLER"] = true
+	names["EXTERNAL"] = true
+
+	for _, item := range listeners {
+		if names[item.Name] {
+			return fmt.Errorf("duplicate of listener name: %s", item.Name)
+		}
+		names[item.Name] = true
+
+		if ports[item.Port] {
+			return fmt.Errorf("duplicate of listener port: %s", item.Port)
+		}
+		ports[item.Port] = true
 	}
+
+	return nil
+}
+
+func editEnvsForListeners(listeners []KafkaListener) map[string]string {
```

Comment:
Let's add unit tests for this function 🙏 

If possible, I think we can combine the functions receiving the listeners into just one. Wdyt?

Reply:
Well, I made them separate by using clean code principles. I think it will be more clear to people editing code after me

Reply:
It's fine, I'm not against it. I simply looked at the code, and saw many steps. Not a blocker, as this is the internal details of the module, not public API